### PR TITLE
openscad: new, 2021.01+git20221104

### PR DESCRIPTION
**Pull request by RedL0tus**
Topic Description
-----------------

Introduce OpenSCAD and its dependencies.

Package(s) Affected
-------------------

`opencsg`: new, 1.5.1
`automatic-components-toolkit`: new, 1.6.0
`lib3mf`: new, 2.2.0
`openscad`: new, 2021.01+git20220514

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by MingcongBai** [comment]
Awaiting test results.

**Comment by RedL0tus** [comment]
image
x86

image
AArch64

File: extra-libs/lib3mf/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,14 @@
+PKGNAME=lib3mf
+PKGSEC=libs
+PKGDES="Implementation of the 3D Manufacturing Format file standard"
+PKGDEP="libzip openssl zlib"
+BUILDDEP="automatic-component-toolkit cmake"
+
+ABTYPE=cmake
```
Comment (MingcongBai) [inline extra-libs/lib3mf/autobuild/defines:7]: Why is it not using Ninja?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Comment by RedL0tus** [comment]
image
Works on AMD64

**Comment by Icenowy** [comment]
图片

Works on ARM64 (Phytium D2000).

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**